### PR TITLE
feat(service):add keccak256,verify in util service

### PR DESCRIPTION
**Pull request by fendouhyz**
**What type of PR is this?**
feature
**What this PR does / why we need it**:
add util service, add keccak256,verify in util service
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: built-in-services/util/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    #[cycles(100_00)]
+    #[read]
+    fn keccak256(
+        &self,
+        ctx: ServiceContext,
+        payload: KeccakPayload,
+    ) -> ServiceResponse<KeccakResponse> {
+        let keccak = HasherKeccak::new();
+
+        let data = hex::decode(payload.hex_str.as_string_trim0x()).unwrap();
```
Comment (huwenchao) [inline built-in-services/util/src/lib.rs:41]: This is a user input error, should return an error instead  of panic.

File: built-in-services/util/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    result: res,
+                };
+                ServiceResponse::<KeccakResponse>::from_succeed(response)
+            }
+
+            _ => ServiceResponse::<KeccakResponse>::from_error(
+                101,
+                "data not valid".to_owned(),
+            )
+        };
```
Comment (huwenchao) [inline built-in-services/util/src/lib.rs:56]: You can omit the `return` and trailing `;` here.

File: built-in-services/util/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    #[cycles(100_00)]
+    #[read]
+    fn verify(
+        &self,
+        ctx: ServiceContext,
+        payload: SigVerifyPayload,
+    ) -> ServiceResponse<SigVerifyResponse> {
+        let data_sig = hex::decode(payload.sig.as_string_trim0x()).unwrap();
+        let data_pk = hex::decode(payload.pub_key.as_string_trim0x()).unwrap();
```
Comment (huwenchao) [inline built-in-services/util/src/lib.rs:67]: Change the `unwrap` here to return error.

File: built-in-services/util/src/tests/mod.rs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    Address, Block, Hash, Proof, Receipt, ServiceContext, ServiceContextParams, SignedTransaction,Hex
+};
+
+use crate::types::{
+    KeccakPayload, KeccakResponse,
+};
+use crate::UtilService;
+
+#[test]
+fn test_hash() {
```
Comment (huwenchao) [inline built-in-services/util/src/tests/mod.rs:26]: Miss `verify` unit test.

File: built-in-services/util/src/lib.rs (hunk 1 above)
Comment (fendouhyz) [reply built-in-services/util/src/lib.rs]: fixed

File: built-in-services/util/src/lib.rs (hunk 2 above)
Comment (fendouhyz) [reply built-in-services/util/src/lib.rs:41]: fixed

File: built-in-services/util/src/lib.rs (hunk 3 above)
Comment (fendouhyz) [reply built-in-services/util/src/lib.rs]: fixed

File: built-in-services/util/src/tests/mod.rs (hunk 4 above)
Comment (fendouhyz) [reply built-in-services/util/src/tests/mod.rs:25]: fixed

**Comment by mohanson** [comment]
Merge?

**Pull request merged**